### DISCUSSION TITLE
Running DockerStorage flow on docker agent

### DISCUSSION THREAD

**Discussion by avlahop** [PrefectHQ/prefect#5192] [Q&A] [answered]
Hello! I am new to prefect and trying to figure out some things, in order to propose it to the company I work for. I am trying to execute a flow using docker agent. I am using prefect server (opensource) and UI. My flow set up is the following

```python
# main.py
from prefect import storage
from prefect.storage import Docker
from prefect.run_configs import DockerRun
from billing.tasks.metrics import calculate_metrics

with Flow("Billing Metric", storage=Docker(python_dependencies=["pymongo"], image_name="billing_flow"), run_config=DockerRun()) as flow:
    calculate_metrics()

flow.register("billing")
```
Billing project is created beforehand. I run the following to register the flow:
```shell
python main.py
```
and I see my flow registered normally toy the UI as well as the stdout. 
I run my agent like
```shell
prefect agent docker start
```

go to prefect UI and choose to start the run (quick run). I get the following message in my agent's stdout:

```shell
[2021-12-02 18:30:00,714] INFO - agent | Deploying flow run 0f2f1469-9722-40da-87a4-373381d08585 to execution environment...
[2021-12-02 18:30:01,383] INFO - agent | Completed deployment of flow run 0f2f1469-9722-40da-87a4-373381d08585
```
but the flow won't start.. I can see that from Prefect UI where the flow stays in `Pending` state. Am I doing something wrong?

Python: v3.8.x
OS: Ubuntu 20.04
docker engine: Docker version 20.10.11, build dea9396

Any help would be highly appreciable!

**Comment by zanieb** [comment, marked as answer]
Hi @avlahop 

You can use the `--show-flow-logs` flag when starting the agent so stream logs from the container which may help you determine what's going on. It's likely that you are encountering the following issue though #4963. If that is the case, you can start your agent with `--network prefect-server` or start your server with `--expose` to resolve the connection problem.

**Reply by avlahop** [reply]
Hello @madkinsz! it worked by starting the server with the `--expose` option. Also can you suggest a good link for structuring a flow project for production using Dockers?

**Reply by zanieb** [reply]
https://github.com/PrefectHQ/prefect/discussions/4042 may be helpful.
